### PR TITLE
EVG-18273: Allow tasks and variants to be generated from any DSI module (not just genny)

### PR DESCRIPTION
**Pull request by mridul-augustine**
…le (not just genny)

### CODE REVIEW

File: dsi/run-from-dsi [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    GENNY_REPO_ROOT="$(pwd -P)"
+    export GENNY_REPO_ROOT
+popd >/dev/null
+
+pushd "$(dirname "$0")/.." >/dev/null
+    WORKLOAD_REPO_ROOT="$(pwd -P)"
+    export WORKLOAD_REPO_ROOT
+popd >/dev/null
+
+"${GENNY_REPO_ROOT}/run-genny" "auto-tasks" "--tasks" "${@}" "--workload-repo" "${WORKLOAD_REPO_ROOT}"
```
Comment (rtimmons) [inline dsi/run-from-dsi:13]: 1. I'm confused about why we need `workload-repo`
2. Similarly, Genny knows the genny repo root by convention. It's always `./src/genny`.
3. What args are being passed in? Not sure I'm following that `--tasks` takes a list of args (`${@}`) versus just the specific task type that's being generated.

File: src/lamplib/src/genny/cli.py [hunk 2]
```diff
@@ -522,14 +522,22 @@ def lint_yaml(ctx: click.Context):
 @click.option(
     "--tasks", required=True, type=click.Choice(["all_tasks", "variant_tasks", "patch_tasks"]),
 )
+@click.option(
+    "-w",
+    "--workload-repo",
```
Comment (rtimmons) [inline src/lamplib/src/genny/cli.py:527]: Seems like genny should just look in all workload repos for genny tasks - I think that's what it's currently doing?

**Comment by mridul-augustine** [comment]
@rtimmons I have completely reworked on the task generation part. I would recommend you read the [docs](https://github.com/mongodb/genny/blob/84d13c9c386ec0329b1e1cf1356cc12ccaedbf27/docs/using.md#creating-a-new-genny-based-workload-repository) to understand the new behavior.

Here is the DSI PR - https://github.com/10gen/dsi/pull/1155
Patch run - https://spruce.mongodb.com/version/63b4d0063627e062b4bb8c16/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

File: docs/using.md [hunk 3]
```diff
@@ -910,6 +911,72 @@ By default, encrypted CRUD operations require a `mongocryptd` daemon to be runni
...
+│   └── run-from-dsi
+└── src
+    └── workloads
+        └── docs
+```
+3. `dsi/run-from-dsi` is the task generator script required by DSI in all workload repositories. This file requires no changes if the repository host has genny workloads. This shell script calls `./run-genny auto-tasks` to generate DSI tasks for all the workloads in the current repository. 
+4. Genny workloads yamls can live in `src/workload/docs` or any directory inside `src/workload`.
+
+<a id="org3aaae9z"></a>
+# Developing a new DSI Task files generator for your custom workload
```
Comment (mridul-augustine) [inline docs/using.md:938]: I guess this should be part of DSI docs? I will move it

**Comment by rtimmons** [comment]
@mridul-augustine will generating variants be a separate PR?

**Comment by mridul-augustine** [comment]
Va

> @mridul-augustine will generating variants be a separate PR?

Variant tasks generation in handle now in [DSI](https://github.com/10gen/dsi/blob/102933480aaf9c51ba21865c4e1c4212bd81d7e1/src/dsi/cli/auto_tasks.py#L187).

Checkout the DSI PR - https://github.com/10gen/dsi/pull/1155 . Let me know if you want to have a discussion.

**Comment by mridul-augustine** [comment]
Changes to be made after the review discussion with @rtimmons 

1. Add schema version for DSI YAML
2. Move the tasks to a seperate tasks section in DSI YAML
3. Warn users if the task names are not unique
4. Move the resposibility of filtering of tasks from DSI to Workload (variant_tasks and patch_tasks)
5. Support versioning for DSI task generator to make sure updates DSI auto_tasks updates are backward comparable.

Let me know if I missed any

File: docs/using.md [hunk 4]
```diff
@@ -31,12 +31,13 @@
         3.  [Default](#org65830c2)
     6.  [Creating an Actor](#org7e6c6bd)
     7.  [Enabling Client-Side Field Level Encryption](#org627591d)
-5.  [Pitfalls](#org3aaae9e)
+5.  [Creating a new Genny based workload repository](#creating-a-new-genny-based-workload-repository)
+6.  [Developing a new DSI Task files generator for your custom workload](#org3aaae9z)
```
Comment (mridul-augustine) [inline docs/using.md:35]: Remove

File: src/lamplib/src/genny/cli.py (hunk 2 above)
Comment (mridul-augustine) [reply src/lamplib/src/genny/cli.py:560]: As discussed. Every workload repo will have its own task generator including genny workloads

**Pull request closed without merging**